qutip/solver/propagator.py

```diff
@@ -138,22 +145,25 @@ class Propagator:
         into a :class:`QobjEvo` with ::
             U = QobjEvo(Propagator(H))
     """
-    def __init__(self, H, c_ops=(), args=None, options=None,
+    def __init__(self, system, *, c_ops=(), args=None, options=None,
                  memoize=10, tol=1e-14):
-        Hevo = QobjEvo(H, args=args)
-        c_ops = [QobjEvo(op, args=args) for op in c_ops]
+        if isinstance(system, McSolver):
```

Comment:
I would prefer it if we didn't have to explicitly check for `McSolver` here so that people who write other solvers can specify themselves whether it supports use in the Propagator.

I'm not sure is `evolve_dm` was the attribute intended for use here instead or not.

Reply:
`evolve_dm` is something I wanted for correlation when doing #1997.
I will use it instead of `isinstance(solver, SeSolver)`